domain/tests/src/Unit/Access/DomainAccessCheckTest.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace Drupal\Tests\domain\Unit\Access;
+
+//use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\domain\Access\DomainAccessCheck;
+use Drupal\Core\Access\AccessResult;
+use Drupal\Tests\UnitTestCase;
+
+/**
+ * Tests the domain record actions.
+ *
+ * @group domain
+ */
+class DomainAccessCheckTest extends UnitTestCase {
+
+  /** @var \Drupal\domain\Access\DomainAccessCheck */
+  private $object;
+
+  /** @var \PHPUnit_Framework_MockObject_MockObject */
+  private $mockNegotiator;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+    $negotiator = $this->getMockBuilder('\Drupal\domain\DomainNegotiator')
+      ->disableOriginalConstructor()
+      ->getMock();
+    $this->mockNegotiator = $negotiator;
+    $this->object = new DomainAccessCheck($negotiator);
+  }
+
+  /**
+   * @param string $path
+   *
+   * @dataProvider providerCheckPathTrue
+   */
+  public function testAppliesTrue($path) {
+    $route = $this->getMockBuilder('\Symfony\Component\Routing\Route')
+      ->disableOriginalConstructor()
+      ->getMock();
+    $route->expects(static::once())->method('getPath')
+      ->willReturn($path);
+    static::assertTrue($this->object->applies($route));
+  }
+
+  /**
+   * @param string $path
+   *
+   * @dataProvider providerCheckPathFalse
+   */
+  public function testAppliesFalse($path) {
+    $route = $this->getMockBuilder('\Symfony\Component\Routing\Route')
+      ->disableOriginalConstructor()
+      ->getMock();
+    $route->expects(static::once())->method('getPath')
+      ->willReturn($path);
+    static::assertFalse($this->object->applies($route));
+  }
+
+  /**
+   * @param string $path
+   *
+   * @dataProvider providerCheckPathTrue
+   */
+  public function testCheckPathTrue($path) {
+    static::assertTrue($this->object->checkPath($path));
+  }
+
+  /**
+   * @param string $path
+   *
+   * @dataProvider providerCheckPathFalse
+   */
+  public function testCheckPathFalse($path) {
+    static::assertFalse($this->object->checkPath($path));
+  }
+
+  /**
+   * @todo Find a way to mock AccessResult::allowedIfHasPermissions to complete test
+   *
+   * If AccessResult::allowedIfHasPermissions cannot be done by mock, then
+   * it will need to be done as a web test case. Maybe an example in comment\Unit\Entity\CommentLockTest
+   */
+  public function testAccess() {
```

Comment:
This is going to be interesting to mock, since the DomainAccessCheck hinges on the status of the domain record and the permissions of the user.

I would expect two mock domain objects to test against -- one where status = 1 and one where status = 0.

I don't know that mocking the Negotiator gets us what we want. It might be easier to mock the DomainLoader?